RecoJets/JetProducers/src/MVAJetPuId.cc

```diff
@@ -0,0 +1,567 @@
+{
+	internalId_ = id;
+}
+
+
+void MVAJetPuId::runMva()
+{
+	if( ! reader_ ) { bookReader();}
+	if(fabs(internalId_.jetEta_) <  5.0) internalId_.mva_ = reader_->EvaluateMVA( tmvaMethod_.c_str() );
+	if(fabs(internalId_.jetEta_) >= 5.0) internalId_.mva_ = -2.;
+	internalId_.idFlag_ = computeIDflag(internalId_.mva_,internalId_.jetPt_,internalId_.jetEta_);
+}
+
+std::pair<int,int> MVAJetPuId::getJetIdKey(float jetPt, float jetEta)
+{
+	int ptId = 0;                                                                                                                                    
+	if(jetPt > 10 && jetPt < 20) ptId = 1;                                                                                 
+	if(jetPt >= 20 && jetPt < 30) ptId = 2;                                                                                 
+	if(jetPt >= 30              ) ptId = 3;                                                                                          
+
+	int etaId = 0;                                                                                                                                   
+	if(fabs(jetEta) > 2.5  && fabs(jetEta) < 2.75) etaId = 1;                                                              
+	if(fabs(jetEta) >= 2.75 && fabs(jetEta) < 3.0 ) etaId = 2;                                                              
+	if(fabs(jetEta) >= 3.0  && fabs(jetEta) < 5.0 ) etaId = 3;                                 
+	return std::pair<int,int>(ptId,etaId);
+}
+
+
+int MVAJetPuId::computeIDflag(float mva, float jetPt, float jetEta)
+{
+	std::pair<int,int> jetIdKey = getJetIdKey(jetPt,jetEta);
+	return computeIDflag(mva,jetIdKey.first,jetIdKey.second);
+}
+
+int MVAJetPuId::computeIDflag(float mva,int ptId,int etaId)
+{
+	int idFlag(0);
+	if(mva > mvacut_[PileupJetIdentifier::kTight ][ptId][etaId]) idFlag += 1 << PileupJetIdentifier::kTight;
+	if(mva > mvacut_[PileupJetIdentifier::kMedium][ptId][etaId]) idFlag += 1 << PileupJetIdentifier::kMedium;
+	if(mva > mvacut_[PileupJetIdentifier::kLoose ][ptId][etaId]) idFlag += 1 << PileupJetIdentifier::kLoose;
+	return idFlag;
+}
+
+PileupJetIdentifier MVAJetPuId::computeMva()
+{
+	runMva();
+	return PileupJetIdentifier(internalId_);
+}
+
+PileupJetIdentifier MVAJetPuId::computeIdVariables(const reco::Jet * jet, float jec, const reco::Vertex * vtx,
+		const reco::VertexCollection & allvtx, double rho,
+		bool calculateMva) 
+{
+
+	typedef std::vector <reco::PFCandidatePtr> constituents_type;
+	typedef std::vector <reco::PFCandidatePtr>::iterator constituents_iterator;
+
+	resetVariables();
+
+	const reco::PFJet * pfjet = dynamic_cast<const reco::PFJet *>(jet);
+
+	if( jec < 0. ) {
+		jec = 1.;
+	}
+
+	constituents_type constituents = pfjet->getPFConstituents();
+
+	reco::PFCandidatePtr lLead, lSecond, lLeadNeut, lLeadEm, lLeadCh, lTrail;
+	std::vector<float> frac, fracCh, fracEm, fracNeut;
+	float cones[4] = { 0.1, 0.2, 0.3, 0.4 };
+	size_t ncones = sizeof(cones)/sizeof(float);
```

Comment:
(this was mentioned in the previous round of comments, in a less specific way though)
use ```constexpr int ncones = 4; ``` and use it in the array definitions.
Also, instead of "float cones[4]" better use ```std::array<float, 4> ```